inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -167,14 +167,20 @@ def calculate_and_execute_sota_upgrade(self, repo: IRepo) -> List:
                                              self._install_check_service)
             if self._repo_type == REMOTE_SOURCE:
                 logger.debug(f"Remote repo URI: {self._uri}")
+                # If uri and signature are provided, pass the uri and signature to the method.
                 if self._uri is not None and self._signature is not None:
                     cmd_list = self.installer.update_remote_source(canonicalize_uri(self._uri),
                                                                    canonicalize_uri(self._signature), repo)
-                elif self._uri is None:
-                    cmd_list = self.installer.update_remote_source(None, None, repo)
+                # If uri is provided but no signature, pass the uri to the method.
+                elif self._uri is not None and self._signature is None:
+                    cmd_list = self.installer.update_remote_source(canonicalize_uri(self._uri),
+                                                                   None, repo)
+                # If signature is provided but no uri, pass the signature to the method.
+                elif self._signature is not None and self._uri is None:
+                    cmd_list = self.installer.update_remote_source(None, canonicalize_uri(self._signature), repo)
```

Comment:
`canonicalize_uri` can't be called on a signature; also, if there is no URI, is this a valid command? there would be nothing for the signature to sign

Reply:
Signature is not the URI. Will remove the `canonicalize_uri` for signature.